### PR TITLE
chore: upgrade Vue to 3.2.47

### LINKED ISSUE #23909: Update to latest Vue version in monorepo
### Current behavior

I tried updating Vue (current on 3.2.31) and I noticed 3.2.32 causes this test to fail:

- Test: https://github.com/cypress-io/cypress/blob/070b3c9ba533b5eed6761ceffdf8636ec2563ae2/packages/app/cypress/e2e/runs.cy.ts#L287
- CI: https://app.circleci.com/pipelines/github/cypress-io/cypress/43742/workflows/31dbcfcd-6da7-4d9d-ad41-e5588c293ad7/jobs/1827401

We rely on an unmounted component (Project Connect Modal) to emit an event `emit('success')` when it is unmounted. This was working in Vue 3.2.31, but apparently unintended, and was changed in this PR to Vue https://github.com/vuejs/core/pull/5679/files which broke us.

Relying on an unmounted component to do anything is definitely not ideal, so we will need to change how that part of the App works.

### Desired behavior

N/A

### Test code to reproduce

N/A

### Cypress Version

N/A

### Node version

N/A

### Operating System

N/A

### Debug Logs

```shell
N/A
```

### Other

_No response_

### PR DESCRIPTION
**Pull request by astone123**
- Closes #23909

### Additional details

This PR upgrades Vue to version `3.2.47` in the monorepo.

### Steps to test

Launch Cypress and do some manual smoke testing to make sure that nothing broke

### How has the user experience changed?

n/a

### PR Tasks

- [ ] Fix types in `mountFragment`?
- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/frontend-shared/cypress/support/mock-graphql/mountFragment.ts [hunk 1]
```diff
@@ -48,6 +48,7 @@ export const registerMountFn = ({ plugins }: MountFnOptions = {}) => {
         },
       })
 
+      // @ts-ignore todo: figure out the correct types
```
Comment (astone123) [inline packages/frontend-shared/cypress/support/mock-graphql/mountFragment.ts:51]: I messed with the type errors in this file for a while... still not sure what needs to change. We already ignore some types in this file (see line 27)

**Comment by lmiller1990** [comment]
Hm, the regression mentioned in https://github.com/cypress-io/cypress/issues/23909#issue-1380480348 is no longer an issue?

**Comment by lmiller1990** [comment]
@astone123 I think https://app.circleci.com/pipelines/github/cypress-io/cypress/51862/workflows/bdae02be-0adc-4fc6-b034-fb4b0b19edcd/jobs/2139937/tests#failed-test-0 is a real regression - might be worth taking a look.

**Review by lmiller1990 — COMMENTED** [review]
Code seems fine, I think we have a regression: https://app.circleci.com/pipelines/github/cypress-io/cypress/51862/workflows/bdae02be-0adc-4fc6-b034-fb4b0b19edcd/jobs/2139937/tests#failed-test-0

**Comment by astone123** [comment]
> Hm, the regression mentioned in [#23909 (comment)](https://github.com/cypress-io/cypress/issues/23909#issue-1380480348) is no longer an issue?

Yeah, I didn't see this regression after the upgrade. I'll look into the other regression

**Comment by astone123** [comment]
@lmiller1990 that test was failing because I updated a default prop to satisfy some new types. I updated the assertion so we should be good now

**Review by mike-plummer — APPROVED** [review]
Odd percy change, but I'll approve assuming that's just flake

**Review by lmiller1990 — APPROVED** [review]
I tried it out, looks good!

I hope we can hit Vue 3.3 soon when it's out. Many nice things landing, like generic components 😎

**Pull request merged**